lib/config/schemas/system.config.json

```diff
@@ -0,0 +1,107 @@
+{
+  "$id": "http://express-gateway.io/models/system.config.json",
+  "type": "object",
+  "properties": {
+    "db": {
+      "type": "object",
+      "properties": {
+        "redis": {
+          "type": "object",
+          "properties": {
+            "emulate": {
+              "type": "boolean",
+              "default": "true"
+            }
+          },
+          "required": [
+            "emulate"
+          ]
+        }
+      },
+      "required": [
+        "redis"
+      ]
+    },
+    "crypto": {
+      "type": "object",
+      "properties": {
+        "cipherKey": {
+          "type": "string",
+          "default": "sensitiveKey"
+        },
+        "algorithm": {
+          "type": "string",
+          "default": "aes256"
+        },
+        "saltRounds": {
+          "type": "number",
+          "default": 10
+        }
+      },
+      "required": [
+        "cipherKey",
```

Comment:
if you have defaults, are they required ?

Reply:
And does it mean that we can comment out those values in system.config ?

Reply:
> if you have defaults, are they required?

Well I'd say yes. They're required fields, but we'll provide good defaults in case you do not provide a value. This is useful when somebody is giving an invalid payload such as 

```javascript
const object = { field: undefined }
``` 

In this case, defaults won't happen (because the field has been provided) but the `required` constraint will mark it as invalid.

> And does it mean that we can comment out those values in system.config ?

Yes. If you have all the default values in place you can technically run the gateway without any config file at all.

Reply:
aha, now I understand about undefined. 

So can we update system config (in `lib` and in CLI templates) to comment out all values that have defaults ?
So it will look like standard apache\nginx\squid or whatever config files. A lot of settings are there with brief explanation,  but they all are disabled

Reply:
You mean like provide the config file with exact the default values but everything commented out, right?

That might sound a good idea. @kevinswiber ?

1.6? Separate release?

Reply:
Let's not block this particular PR waiting for the detailed comments.  Make an issue for it, though, and we can try to get it done for 1.6.